### PR TITLE
Manage members in app

### LINKED ISSUE #779: Remove the notification "The contacts app is required to manage members"
**Is your feature request related to a problem? Please describe.**

I provide access to the App Contacts through a group because it's an unnecessary module for most users. Only admins of the "Collective" have access to it. As a result, all users who don't have access to the App Contacts constantly encounter the error message "The contacts app is required to manage members," which is disruptive.

Screenshot 2023-07-11 095305

**Describe the solution you'd like**

Remove the existing notification. Add the text "You don't have access to the Contacts App. Please contact the Administrator" to the Collective card.

Screenshot 2023-07-11 100551

**Additional context**

If the proposed solution is suitable, I can implement it.

### PR DESCRIPTION
**Pull request by mejo-**
### 📝 Summary

This PR adds functionality to manage members for a collective inside the Collectives app. This removes our dependency on the Contacts app to do so.

It's possible to add and remove members and promote/demote their membership level.

* Resolves: #308
* Resolves: #244

#### 🖼️ Screenshots

Searching for members | Changing membership level
---|---
grafik | grafik

#### 🖼️ Screencast

| 🏡 Walkthrough creating a collective and managing its members |
|---|
| [recording1.webm](https://github.com/nextcloud/collectives/assets/3582805/4ef9d177-205e-4937-98d0-c272afd35e49) |

### 🚧 TODO

- [x] Cypress tests
- [x] Don't allow to delete circle/group if current admin is member through it
- [x] Always show option to leave collective (also to admins)
- [x] Remove link to manage members in contacts
- [x] Remove dependency checks on contacts app (and close #779)
- [x] Filtering out member groups doesn't work (groups have type 16 in circle response)

### 🏁 Checklist

- [x] Code is properly formatted (`npm run lint` / `npm run stylelint` / `composer run cs:check`)
- [x] [Sign-off message](https://probot.github.io/apps/dco/) is added to all commits
- [x] Tests (unit, integration and/or end-to-end) passing and the changes are covered with tests
- [x] Documentation ([README](https://github.com/nextcloud/collectives/blob/main/README.md) or [documentation](https://github.com/nextcloud/collectives/blob/main/docs/)) has been updated or is not required

### CODE REVIEW

**Comment by mejo-** [comment]
@nextcloud/designers I'd appreciate feedback from you :blush: In particular on the icons used for admin/moderator/member levels and on the way the loading indicator icon is used.

**Review by marcoambrosini — APPROVED** [review]
From a design perspective, I'm sorry but I don't have any constructive feedback to share. It looks very neat and easy to use, congrats!! :)

File: cypress/e2e/collective-members.spec.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+			.find('.action-item__menutoggle')
+			.click({ force: true })
+		cy.get('button.action-button')
+			.contains('Manage members')
+			.click()
+	})
+
+	describe('Manage members', function() {
+		it('Allows to add members', function() {
+			// const addMembers = ['alice', 'Bobs Group']
```
Comment (juliushaertl) [inline cypress/e2e/collective-members.spec.js:48]: Any reason to leave this around? Shall we have a separate test for the groups?

**Review by juliushaertl — COMMENTED** [review]
Looks very nice, code also looks good, just a few commments from testing:

- [x] The search results are not keyboard accessible
- [ ] The scroll in the modal seems a bit odd with two scroll areas:
   
Screenshot 2023-07-25 at 09 30 04

- [ ] Maybe we can hide the "Add users, groups  or circles ..." heading if the search is not triggered, the placeholder hint to search to add groups seems enough and the members also don't have a heading in the default state

Screenshot 2023-07-25 at 09 32 39

File: cypress/e2e/collective-members.spec.js (hunk 1 above)
Comment (mejo-) [reply cypress/e2e/collective-members.spec.js:48]: Yes, I want to switch back to test with a group as well as soon as I resolved the following problem: for now, groups that are part of members already are not filtered out of the search results. The problem is that groups have type `16` (`TYPE_GROUP`) in circle response.

**Comment by mejo-** [comment]
>     Maybe we can hide the "Add users, groups or circles ..." heading if the search is not triggered, the placeholder hint to search to add groups seems enough and the members also don't have a heading in the default state

@juliushaertl I would prefer to keep it. Talk behaves the same in the participants tab and I think its good to emphasize that you can search for new members to add.

**Review by juliushaertl — APPROVED** [review]

**Comment by max-nextcloud** [comment]
Looking at the screencast it seems that people, circles and groups that are already in the collective are still listed in the sections to add members. I tried in talk and it seems to filter them out. I think that would be good.

**Comment by mejo-** [comment]
> Looking at the screencast it seems that people, circles and groups that are already in the collective are still listed in the sections to add members. I tried in talk and it seems to filter them out. I think that would be good.

This is already fixed in the PR, but I didn't update the screencast :relaxed:

**Comment by nimishavijay** [comment]
Looks super nice! Only some minor points :)

- We generally don't use the "light" font weight very much, instead we use the regular font weight in `--color-text-maxcontrast`
- We could switch the icons for admin and moderator since the admin role has the most permissions
- Since there are 2 levels of permissions, there could be an simple explanation in a subline for all the roles to avoid confusion between admin and moderators
  - Admin: Manage collective members, view, edit, share and delete collectives
  - Moderators: View collectives, edit and share collectives if permitted
  - Members: View collectives, edit and share collectives if permitted

**Comment by mejo-** [comment]
Thanks @nimishavijay. I already changed the font color to `--color-text-maxcontrast` and switched the icons. For the level explanations this is a bigger thing and I'd like to postpone it, as I either would have to reimplement or extend NcActionButton.

**Pull request merged**